frontend/src/components/user/Heatmap.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react';
+import CalendarHeatmap from 'react-calendar-heatmap';
+import 'react-calendar-heatmap/dist/styles.css';
+import { type User } from '../../types/users/users';
+import './Heatmap.css';
+import Tooltip from 'react-tooltip';
+import { Flex, Text } from '@chakra-ui/react';
+
+interface HeatmapProps {
+  user: User;
+}
+
+interface AnsweredQuestion {
+  id: number;
+  userId: number;
+  questionId: number;
+  complexity: string;
+  category: string[];
+  answeredAt: string;
+}
+
+const getMonth = (monthNumber: number): string => {
+  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+  return months[monthNumber - 1];
+};
+
+const HeatmapComponent: React.FC<HeatmapProps> = ({ user }) => {
+  const [activityData, setActivityData] = useState<Array<{ date: string; count: number }>>([]);
+  const [totalCount, setTotalCount] = useState(0);
+  useEffect(() => {
+    const fetchData = async (): Promise<void> => {
+      try {
+        const response = await fetch(`http://localhost:8000/api/user/get-answered-questions/${user.id}`);
```

Comment:
might be better to use env var REACT_APP_USER_SERVICE_BACKEND_URL here for the backend URL, else it might not work on prod(?)